### ISSUE TITLE
Embedded version of gtest/gmock is buggy with recent compilers

### ISSUE THREAD

**Issue by rleigh-codelibre** [jbeder/yaml-cpp#493]
Fails on Ubuntu 15.04 with GCC 6.3.0 and FreeBSD 11 with clang 3.8.0.  Probably C++-11 by default for both?

```
% gdb test/run-tests
…
(gdb) run
…
[----------] 10 tests from EncodingTest
[ RUN      ] EncodingTest.UTF8_noBOM

Program received signal SIGSEGV, Segmentation fault.
testing::internal::FunctionMockerBase<void (YAML::Mark const&)>::InvokeWith(std::tuple<YAML::Mark const&> const&) (
    args=std::tuple containing = {...}, this=0x555555c1b490) at /data/code/yaml-cpp/test/gmock-1.7.0/include/gmock/gmock-spec-builders.h:1530
1530            this->UntypedInvokeWith(&args))->GetValueAndDelete();
(gdb) bt
#0  testing::internal::FunctionMockerBase<void (YAML::Mark const&)>::InvokeWith(std::tuple<YAML::Mark const&> const&) (
    args=std::tuple containing = {...}, this=0x555555c1b490) at /data/code/yaml-cpp/test/gmock-1.7.0/include/gmock/gmock-spec-builders.h:1530
#1  testing::internal::FunctionMocker<void (YAML::Mark const&)>::Invoke(YAML::Mark const&) (a1=..., this=0x555555c1b490)
    at /data/code/yaml-cpp/test/gmock-1.7.0/include/gmock/gmock-generated-function-mockers.h:97
#2  YAML::MockEventHandler::OnDocumentStart (this=this@entry=0x555555c1b488, gmock_a1=...) at /home/rleigh/code/yaml-cpp/test/mock_event_handler.h:10
#3  0x000055555586f25f in YAML::SingleDocParser::HandleDocument (this=this@entry=0x7fffffffdbb0, eventHandler=...)
    at /home/rleigh/code/yaml-cpp/src/singledocparser.cpp:32
#4  0x000055555585457d in YAML::Parser::HandleNextDocument (this=this@entry=0x7fffffffdd00, eventHandler=...)
    at /home/rleigh/code/yaml-cpp/src/parser.cpp:39
#5  0x00005555556255f5 in YAML::HandlerTest::Parse (
    example="- |\n  !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~\n- |\n  ¡¢£¤¥¦§¨©ª«¬­®¯°±²³´µ¶·¸¹º»¼½¾¿ÀÁÂÃÄÅÆÇÈÉÊËÌÍ", <incomplete sequence \303>..., this=0x555555c1b470)
    at /home/rleigh/code/yaml-cpp/test/handler_test.h:17
#6  YAML::(anonymous namespace)::EncodingTest::Run (this=0x555555c1b470) at /home/rleigh/code/yaml-cpp/test/integration/encoding_test.cpp:105
#7  0x00005555558a6ef4 in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void> (location=0x5555558fe3a9 "the test body", 
    method=<optimised out>, object=<optimised out>) at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:2078
#8  testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void> (object=object@entry=0x555555c1b470, method=<optimised out>, 
    location=location@entry=0x5555558fe3a9 "the test body") at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:2114
#9  0x000055555589ef4a in testing::Test::Run (this=this@entry=0x555555c1b470) at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:2151
#10 0x000055555589f13d in testing::Test::Run (this=0x555555c1b470) at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:2142
#11 testing::TestInfo::Run (this=0x555555bd4530) at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:2326
#12 0x000055555589f2c5 in testing::TestInfo::Run (this=<optimised out>) at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:2301
#13 testing::TestCase::Run (this=0x555555bd4620) at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:2444
#14 0x000055555589f77d in testing::TestCase::Run (this=<optimised out>) at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:4353
#15 testing::internal::UnitTestImpl::RunAllTests (this=0x555555bcbd70) at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:4315
#16 0x00005555558a73a4 in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (
    location=0x5555558fd9f8 "auxiliary test code (environments or event listeners)", method=<optimised out>, object=<optimised out>)
    at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:2078
#17 testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (object=0x555555bcbd70, method=<optimised out>, 
    location=location@entry=0x5555558fd9f8 "auxiliary test code (environments or event listeners)")
    at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:2114
#18 0x000055555589fab9 in testing::UnitTest::Run (this=0x555555bb9020 <testing::UnitTest::GetInstance()::instance>)
    at /home/rleigh/code/yaml-cpp/test/gmock-1.7.0/gtest/src/gtest.cc:3926
#19 0x00005555555ef45f in RUN_ALL_TESTS () at /data/code/yaml-cpp/test/gmock-1.7.0/gtest/include/gtest/gtest.h:2288
#20 main (argc=<optimised out>, argv=<optimised out>) at /home/rleigh/code/yaml-cpp/test/main.cpp:5
```

The bug does not appear to manifest with gtest 1.8.0.

I'll have a patch for this shortly, which will also address https://github.com/jbeder/yaml-cpp/issues/488

**Comment by keith-bennett-gbg** [comment]
https://github.com/jbeder/yaml-cpp/issues/478

**Comment by jbeder** [comment]
Fixed by #494.
